server/src/main/java/io/crate/metadata/functions/Signature.java

```diff
@@ -327,6 +344,11 @@ public Signature(StreamInput in) throws IOException {
         int enumElements = in.readVInt();
         features = Collections.unmodifiableSet(EnumSets.unpackFromInt(enumElements, Scalar.Feature.class));
         bindingInfo = null;
+        if (in.getVersion().onOrAfter(Version.V_5_8_0)) {
+            nullability = in.readEnum(Scalar.Nullability.class);
+        } else {
+            nullability = Scalar.Nullability.DEFAULT;
+        }
```

Comment:
Couldn't we keep the internal representation the same - keep using the features, and make this only a constructor change?

If I remember right the reason to go with features in the initial PR was to reduce the memory footprint.

Reply:
Apologies, it was my suggestion to make in another enum, and it was 100% naming driven.
(To have a nicer default value: `Nullability.DEFAULT`)